### PR TITLE
feat: value to conditionally render ExternalSecrets instead of Secrets

### PR DESCRIPTION
**Pull request by joshuasimon-taulia**
the current external secrets implementation is coupled with the `jx secrets` logic in the bootjob Makefile. for charts that we own such as https://github.com/jenkins-x-charts/jxboot-helmfile-resources/tree/main/charts/jxboot-helmfile-resources that contain secrets objects which will obviously be converted to `ExternalSecrets`, how do you feel about removing a layer of abstraction and creating the `ExternalSecret` object directly in the helm chart instead of relying on the` jx secret` [schema](https://github.com/jenkins-x/jx3-versions/blob/bc26c537be05c250c6a584c131e24b3ebf10f98e/charts/jxgh/jxboot-helmfile-resources/secret-schema.yaml#L230-L292)? this would also make it much easier to swap out the bootjob for something like argoCD

### CODE REVIEW

File: charts/jxboot-helmfile-resources/templates/jx-basic-auth-htpasswd-secret.yaml [hunk 1]
```diff
@@ -1,4 +1,29 @@
 {{- if .Values.basicAuthSecrets.enabled }}
+apiVersion: kubernetes-client.io/v1
```
Comment (gazal-k) [inline charts/jxboot-helmfile-resources/templates/jx-basic-auth-htpasswd-secret.yaml:2]: I would suggest moving to ESO, https://external-secrets.io/v0.7.2/. https://github.com/external-secrets/kubernetes-external-secrets has been deprecated.

Comment (joshuasimon-taulia) [reply charts/jxboot-helmfile-resources/templates/jx-basic-auth-htpasswd-secret.yaml:2]: i think there's still some work required on this front. https://github.com/jenkins-x-plugins/jx-secret/pull/387

File: charts/jxboot-helmfile-resources/templates/jx-basic-auth-htpasswd-secret.yaml [hunk 2]
```diff
@@ -1,4 +1,29 @@
...
+kind: ExternalSecret
+metadata:
+  name: jx-basic-auth-htpasswd
+  namespace: jx
+  annotations:
+    secret.jenkins-x.io/schema-object: {{`'{"name":"jx-basic-auth-htpasswd","properties":[{"name":"auth","question":"the htpasswd format basic auth for Ingress","help":"The htpasswd encoded user and password for basic auth ingresses","template":"{{ htpasswdExtSecret \"jx-basic-auth-user-password\" \"username\" \"password\" }}","onlyTemplateIfBlank":true,"retry":true}]}'`}}
+  labels:
+    gitops.jenkins-x.io/pipeline: 'namespaces'
+spec:
+  backendType: gcpSecretsManager
```
[the remark this answers is not in the document]
Comment (joshuasimon-taulia) [reply charts/jxboot-helmfile-resources/templates/jx-basic-auth-htpasswd-secret.yaml:13]: [fixed](https://github.com/jenkins-x-charts/jxboot-helmfile-resources/pull/662/commits/db36520ad55e4df3eb033100e279b53a1dad324c). please set PR merge type to squash

**Comment by haysclark** [comment]
I just include ExternalSecrets in my application templates. What gets messy is mapping external secrets in a Multi-cluster repo setup, as the `Project` and `ProjectId` can't be hard-coded and need to be dynamic, depending on the environment.

**Comment by msvticket** [comment]
Since we are about to move to ESO this is not really a suitable thing to do now. If this had been merged it would have become another hurdle for that shift. So, no, I don't think this is a good idea.

@tomhobson

**Pull request closed without merging**